reconcile/ecr_mirror.py

```diff
@@ -0,0 +1,138 @@
+import base64
+import logging
+
+from sretoolbox.container import Image
+from sretoolbox.container import Skopeo
+from sretoolbox.container.skopeo import SkopeoCmdError
+
+import reconcile.utils.threaded as threaded
+
+from reconcile import queries
+from reconcile.utils.aws_api import AWSApi
+from reconcile.utils.secret_reader import SecretReader
+
+
+QONTRACT_INTEGRATION = 'ecr-mirror'
+LOG = logging.getLogger(__name__)
+
+
+class EcrMirror:
+    def __init__(self, instance, dry_run):
+        self.dry_run = dry_run
+        self.instance = instance
+        self.settings = queries.get_app_interface_settings()
+        self.secret_reader = SecretReader(settings=self.settings)
+        self.skopeo_cli = Skopeo(dry_run)
+        self.error = False
+
+        identifier = instance['identifier']
+        account = instance['account']
+        region = instance.get('region')
+
+        self.aws_cli = AWSApi(thread_pool_size=1,
+                              accounts=[
+                                  self._get_aws_account_info(account)
+                              ],
+                              settings=self.settings,
+                              init_ecr_auth_tokens=True)
+
+        self.aws_cli.map_ecr_resources()
+
+        self.ecr_uri = self._get_image_uri(
+            account=account,
+            repository=identifier,
+        )
+        if self.ecr_uri is None:
+            self.error = True
+            LOG.error(f"Could not find the ECR repository {identifier}")
+
+        self.ecr_username, self.ecr_password = self._get_ecr_creds(
+            account=account,
+            region=region,
+        )
+        self.ecr_auth = f'{self.ecr_username}:{self.ecr_password}'
+
+        self.image_username = None
+        self.image_password = None
+        self.image_auth = None
+        push_secret = self.instance['mirror']['pullCredentials']
+        if push_secret is not None:
```

Comment:
why this if?

Reply:
if it so happens that push_secret is None, shouldn't we raise an exception?

Reply:
it's pull secret... typo. Repo to pull from might be public.

Reply:
nevermind, I understand it now